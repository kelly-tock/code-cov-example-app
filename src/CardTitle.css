.CardTitle {
  padding: 16px;
  border-bottom: 1px solid gray;
}
