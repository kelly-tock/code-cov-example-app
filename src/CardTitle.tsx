import React, { FunctionComponent } from "react";
import clsx from "clsx";
import "./CardTitle.css";

interface CardTitleProps {
  title: string;
  subtitle?: string;
}

export const CardTitle: FunctionComponent<CardTitleProps> = ({
  title,
  subtitle,
}) => {
  return (
    <div className="CardTitle">
      <h2>{title}</h2>
      {subtitle ? <p>{subtitle}</p> : null}
    </div>
  );
};
