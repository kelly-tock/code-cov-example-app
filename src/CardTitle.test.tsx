import React from "react";
import { render, screen } from "@testing-library/react";
import { CardTitle } from "./CardTitle";

test("matches default snap", () => {
  const { container } = render(<CardTitle title="title" />);
  expect(container.firstChild).toMatchSnapshot();
});

test("matches default snap with subtitle", () => {
  const { container } = render(<CardTitle title="title" subtitle="subtitle" />);
  expect(container.firstChild).toMatchSnapshot();
});
