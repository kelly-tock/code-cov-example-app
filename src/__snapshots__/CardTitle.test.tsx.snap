// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches default snap 1`] = `
<div
  class="CardTitle"
>
  <h2>
    title
  </h2>
</div>
`;

exports[`matches default snap with subtitle 1`] = `
<div
  class="CardTitle"
>
  <h2>
    title
  </h2>
  <p>
    subtitle
  </p>
</div>
`;
